src/models/ProcurementRequests.php

```diff
@@ -0,0 +1,142 @@
+<?php
+
+/**
+ * @author Nicolas CARPi <nico-git@deltablot.email>
+ * @copyright 2024 Nicolas CARPi
+ * @see https://www.elabftw.net Official website
+ * @license AGPL-3.0
+ * @package elabftw
+ */
+
+declare(strict_types=1);
+
+namespace Elabftw\Models;
+
+use Elabftw\Elabftw\Db;
+use Elabftw\Elabftw\ProcurementRequestParams;
+use Elabftw\Enums\Action;
+use Elabftw\Enums\Currency;
+use Elabftw\Enums\ProcurementState;
+use Elabftw\Exceptions\ImproperActionException;
+use Elabftw\Interfaces\RestInterface;
+use Elabftw\Services\TeamsHelper;
+use Elabftw\Traits\SetIdTrait;
+use PDO;
+use RuntimeException;
+
+/**
+ * Procurement requests are purchase orders in a team
+ */
+class ProcurementRequests implements RestInterface
+{
+    use SetIdTrait;
+
+    protected Db $Db;
+
+    public function __construct(protected Teams $Teams, ?int $id = null)
+    {
+        $this->Db = Db::getConnection();
+        $this->setId($id);
+    }
+
+    public function readAll(): array
+    {
+        $sql = "SELECT CONCAT(users.firstname, ' ', users.lastname) AS requester_fullname, pr.id, pr.created_at, pr.team, pr.requester_userid, pr.entity_id, pr.qty_ordered, pr.qty_received, pr.body, pr.quote, pr.email_sent, pr.state, items.title AS entity_title, pr.qty_ordered * items.proc_price_tax AS total, items.proc_currency
```

Comment:
it helps on the long run if the individual columns are on separate rows or at least not more than 3-4 especially when you use a function like `CONCAT` and/or `AS`

why is there a `*` in the select? do we need all columns?

Reply:
line splitted. The `*` is the multiplication of two columns to get the total price.